src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementArray;
+    private int size;
+    private int capacity;
+
+    public ArrayList() {
+        capacity = DEFAULT_CAPACITY;
+        elementArray = (T[]) new Object[capacity];
+    }
+
+    public T[] elementArrayPlusSize(T[] elementArray, int increasingInSizeByInt) {
+        int futureSize = size + increasingInSizeByInt;
+        while (futureSize >= capacity) {
+            capacity += (capacity >> 1);
+        }
+        T[] arrayPlusSize = (T[]) new Object[capacity];
+        System.arraycopy(elementArray, 0, arrayPlusSize, 0, size);
+        return arrayPlusSize;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            elementArray = elementArrayPlusSize(elementArray, 1);
+        }
+        elementArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            if (size == capacity) {
+                elementArray = elementArrayPlusSize(elementArray, 1);
+            }
+            size++;
+            System.arraycopy(elementArray, index, elementArray, index + 1, size - index - 1);
+            elementArray[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("this index is not correction");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
```

Comment:
You can use add method
example:
```
    @Override
    public void addAll(List<T> list) {
        for(int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
    }
```

Reply:
use this approach, no need to handle grow here - there's logic for that in add method already. addAll is just a convenient way to add multiple times, that's it